volatility3/framework/symbols/linux/utilities/modules.py

```diff
@@ -346,52 +268,30 @@ def run_modules_scanners(
         Returns:
             Dictionary mapping each plugin to its corresponding result
         """
-
-        module_gatherers = {
-            Modules.source_kernel_identifier: cls.get_kernel_module_info,
-            Modules.source_lsmod_identifier: cls._get_list_modules,
-            Modules.source_sysfs_identifier: cls._get_sysfs_modules,
-        }
+        # Throws ValueError if invalid gatherers sent in
+        Modules._validate_gatherers(caller_wanted_gatherers)
 
         kernel = context.modules[kernel_module_name]
 
         address_mask = context.layers[kernel.layer_name].address_mask
 
-        wanted_sources = Modules._validated_sources(caller_wanted_sources)
-
-        run_results = {}
-
-        run_hidden_modules = False
-
-        # Special case hidden modules since it gathers modules on its own
-        if Modules.source_hidden_identifier in wanted_sources:
-            run_hidden_modules = True
-            wanted_sources.remove(Modules.source_hidden_identifier)
-
-        # Walk each source, gathering modules
-        for wanted_source in wanted_sources:
-            run_results[wanted_source] = []
+        run_results: Dict[ModuleGathererInterface, List[ModuleInfo]] = {}
```

Comment:
Might be useful for a `ModuleGathererInterface` to have a name, so that the results can be (uniquely) referred to by something other than a class...

Reply:
hmm then we end up duplicating all the identifers.

Reply:
Are there identifiers anywhere?  At the moment it's would just be `__class__.__name__` to figure out something you could print to show where it came from?  Seems handy to be able to say `"hidden_modules"`  rather than `"ModulesGathererHidden"` for example?

Reply:
This would also get weird as the plugins would have to know the identiifer plus the class name unless we sent back some mixed data structures. Look at modxview, its very easy to distinguish these.

Reply:
No, no, you can index by the class, but just have the class tagged with a string identifier.  You never even have to access it if you don't want, but given you don't know exactly which gatherers may be included, you might want to let plugins get a human readable string, rather than just handing them back a class...

Reply:
Ok do you mean tagged like adding a property in the constructor, then callers could do `.name`? If that I don't mind that as much. Although every gatherer would have to override __init__

Reply:
No, just a class attribute, one per scanner only, not per instance.  I think this has been sorted, so marking as resolved.